### ISSUE TITLE
Reading file ignores all other transforms

### ISSUE THREAD

**Issue by BrianGenisio** [javoire/browserify-ng-html2js#23]
The way this transform is written, it ignores the [data](https://github.com/javoire/browserify-ng-html2js/blob/master/lib/index.js#L37) that it is passed and instead just [reads the file](https://github.com/javoire/browserify-ng-html2js/blob/master/lib/index.js#L43) directly.  This bypasses any other transforms that may occur on the content before it gets to this transform.

In my case, my HTML files all have a BOM at the beginning.  I have been spinning for hours, trying to figure out why using [strip-bomify](https://www.npmjs.com/package/strip-bomify) isn't working, and I tracked it down to this mechanism.  It IS working, but this transform is ignoring my previous transform!

An alternative approach would be to add BOM stripping right into this transform.  After reading the content: `content = content.replace(/^\ufeff/g, '');`

Thoughts?

**Comment by BrianGenisio** [comment]
PR: https://github.com/javoire/browserify-ng-html2js/pull/24

**Comment by javoire** [comment]
Interesting. Do you think it could be rewritten to use the `data` instead of reading the file directly?

**Comment by javoire** [comment]
Now that I look at it, it's kinda strange that the file is read even though the `data` stream should be done, and thus contain the file contents needed. Might have a look into this at some point

**Comment by BrianGenisio** [comment]
Yeah, if you look at other Browserify plugins, they all use the `data` so that they can be streamed together.  I've never seen one that reads the file.  The file is handed to the plugin for two main reasons: filtering (you don't want to templatize a *.JS file) and logging.  The intention is to process the data as it comes though.  

In my case, I can run the `strip-bomify` plugin before this one to solve my problem... but only if it behaves this way.  

I created the other PR because I didn't know if there was a reason it was doing it this way.  But if the answer is "this is just the way I built it", then yeah, I think it should use the `data` instead ;)
